tools/RAiDER/processWM.py

```diff
@@ -140,16 +136,33 @@ def prepareWeatherModel(
         del weather_model
 
 
-def checkBounds(weather_model, outLats, outLons):
-    '''Check the bounds of a weather model'''
-    ds = xr.load_dataset(weather_model.files[0])
-    coords = ds.coords  # coords is dict-like
-    keys = [k for k in coords.keys()]
-    xc = coords[keys[0]]
-    yc = coords[keys[1]]
-    lat_bounds = [yc.min(), yc.max()]
-    lon_bounds = [xc.min(), xc.max()]
-    self_extent = lat_bounds + lon_bounds
-    in_extent = weather_model._getExtent(outLats, outLons)
+def checkContainment(weather_model: weatherModel,
+                     outLats: np.ndarray,
+                     outLons: np.ndarray) -> bool:
+    '''Check if weather model contains bounding box of outLats and outLons'''
+    weather_model_path = weather_model.files[0]
+    with rasterio.open(f'netcdf:{weather_model_path}') as ds:
+        datasets = ds.subdatasets
+
+    with rasterio.open(datasets[0]) as ds:
```

Comment:
I definitely think this could fail. This check happens to the original weather model data (as opposed to the processed) so we don't know what variables exist and whether they are always the first. My opinion would be to use either gdal, rasterio, or xarray to query the coordinates specifically (e.g. the "coords" attribute in xarray). I will post a time test results later.

Reply:
I just want to clarify a few points about when/how this will fail. 

a) This only selects _geo-referenced_ variables in an netcdf files. For example, it would ignore "levels" in an HRES processed weather model file
b) If there are multiple subgroups with different coordinate reference frames then this may fail if the reference frames have different bounding box - in this case you couldn't open the entire dataset with xarray; you would have to specify the group.

There is a third instance (coming from aria) in which a non-raster, georeferenced object is included in the netcdf. I don't think this will be a problem for the data we are considering.

Reply:
@cmarshak ah I see! I was essentially thinking about a levels or similar dataset coming from the weather model file, but evidently that is not an issue. I think this could work fine then.